### PR TITLE
Add V1GetterToV2 Recipe

### PR DESCRIPTION
**Pull request by davidh44**
## Motivation and Context

Transform v1 getters to v2 fluent style, removing the `"get"`

## Testing

added unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [ ] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: migration-tool/src/test/java/software/amazon/awssdk/migration/recipe/V1GetterToV2Test.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    public void defaults(RecipeSpec spec) {
+        spec.recipes(new ChangeSdkType(), new NewClassToBuilderPattern(), new V1GetterToV2());
+        spec.parser(Java8Parser.builder().classpath("sqs",
+                                                    "aws-java-sdk-sqs",
+                                                    "sqs",
+                                                    "sdk-core"));
+    }
+
+    @Test
+    @EnabledOnJre({JRE.JAVA_8})
```
[the remark this answers is not in the document]
Comment (davidh44) [reply migration-tool/src/test/java/software/amazon/awssdk/migration/recipe/V1GetterToV2Test.java:24]: rewriteRun() is OpenRewrite's method of testing

https://docs.openrewrite.org/authoring-recipes/recipe-testing

The other recipes are tested this way:
https://github.com/aws/aws-sdk-java-v2/blob/feature/master/migration-tool/migration-tool/src/test/java/software/amazon/awssdk/migration/recipe/ChangeSdkTypeTest.java

[the remark this answers is not in the document]
Comment (joviegas) [reply migration-tool/src/test/java/software/amazon/awssdk/migration/recipe/V1GetterToV2Test.java:24]: In that case I guess its fine.
Thanks for sharing this.
Will check with @zoewangg  if we can move the Hardcoded code to a resourceFiles via a refactoring PR 
```java
spec -> spec.parser(Java8Parser.builder().classpath("aws-java-sdk-sqs")),
            java(StringContentFromResourceFile
```

File: migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/V1GetterToV2.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+            if (selectType == null) {
+                return method;
+            }
+
+            String methodName = method.getSimpleName();
+            JavaType.FullyQualified fullyQualified = TypeUtils.asFullyQualified(selectType);
+
+            if (NamingUtils.isGetter(methodName)) {
+                methodName = NamingUtils.removeGet(methodName);
```
Comment (joviegas) [inline migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/V1GetterToV2.java:67]: How to we make sure it does not remove getters which are not associated with v1 variables ?
Can we add a test cases where we have instances as below

```
Locale locale = Locale.getDefault();
String path = System.getenv("PATH");
String className = String.class.getName(); 
```

Comment (davidh44) [reply migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/V1GetterToV2.java:71]: Added check for model class and additional test cases

**Review by joviegas — APPROVED** [review]

**Pull request merged**